src/main/java/gameframework/gui/GameStatusBar.java

```diff
@@ -1,50 +1,45 @@
 package gameframework.gui;
 
-import gameframework.base.ObservableValue;
-
 import java.awt.Container;
 import java.awt.GridBagLayout;
-import java.awt.Label;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Observable;
 import java.util.Observer;
 
 import javax.swing.JPanel;
 
-public class GameStatusBar implements Observer {
 
-	protected Label lifeText, scoreText;
-	protected Label lifeValue, scoreValue;
+public class GameStatusBar implements Observer {
 
-	protected final ObservableValue<Integer> score;
-	protected final ObservableValue<Integer> life;
-
-	public GameStatusBar(ObservableValue<Integer> score,
-			ObservableValue<Integer> life) {
-		this.score = score;
-		this.life = life;
-		score.addObserver(this);
-		life.addObserver(this);
-	}
+	protected ArrayList<GameStatusBarElement> elements = new ArrayList<GameStatusBarElement>();
 
 	public Container getContainer() {
-		JPanel c = new JPanel();
+		JPanel container = new JPanel();
 		GridBagLayout layout = new GridBagLayout();
-		c.setLayout(layout);
-		lifeText = new Label("Lives:");
-		scoreText = new Label("Score:");
-		lifeValue = new Label();
-		scoreValue = new Label();
-		c.add(lifeText);
-		c.add(lifeValue);
-		c.add(scoreText);
-		c.add(scoreValue);
+		container.setLayout(layout);
+		
+		for(GameStatusBarElement element : elements){
+			container.add(element.getElementText());
+			container.add(element.getElementValue());
+		}
+		
 		update();
-		return c;
+		return container;
+	}
+	
+	public void addElement(GameStatusBarElement newElement) {
+		this.elements.add(newElement);
+	}
+	
+	public void addAllElement(List<GameStatusBarElement> newElements) {
+		this.elements.addAll(newElements);
 	}
 
 	public void update() {
-		lifeValue.setText(Integer.toString(life.getValue()));
-		scoreValue.setText(Integer.toString(score.getValue()));
+		for(GameStatusBarElement element : elements){
+			element.getElementValue().setText(Integer.toString(element.getElement().getValue()));
```

Comment:
You might want to delegate the computation of the String representation to the element itself. That way, the code becomes simpler and we open the door to non-integer elements.